p-token/src/entrypoint-runtime-verification.rs

```diff
@@ -1815,19 +1815,20 @@ pub fn test_process_close_account(accounts: &[AccountInfo; 3]) -> ProgramResult
         } else if accounts[1].key() != &INCINERATOR_ID {
             assert_eq!(result, Err(ProgramError::InvalidAccountData));
             return result;
-        } else if dst_init_lamports.checked_add(src_init_lamports).is_none() {
+        }
+        if dst_init_lamports.checked_add(src_init_lamports).is_none() {
             assert_eq!(result, Err(ProgramError::Custom(14)));
             return result;
         }
+        assert!(result.is_ok());
 
         // Validate owner falls through to here if no error
         assert_eq!(accounts[0].lamports(), 0);
         assert_eq!(
             accounts[1].lamports(),
             dst_init_lamports + src_init_lamports
         );
-        assert_eq!(accounts[0].data_len(), 0); // TODO: More sol_memset stuff?
-        assert!(result.is_ok());
+        // assert_eq!(accounts[0].data_len(), 0); // Solana-RT only
```

Comment:
Would it be better to put this behind the feature flag that is for this the same as is used in pinocchio?
```suggestion
        #[cfg(any(target_os = "solana", target_arch = "bpf"))]
        assert_eq!(accounts[0].data_len(), 0); // Solana-RT only
```